muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/workspaces.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import {StateType} from '~/reducers';
+import {connect, Dispatch} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import ApplicationList, { ApplicationListItem } from '~/components/general/application-list';
+import Workspace from './workspaces/workspace';
+import {i18nType} from '~/reducers/base/i18n';
+import {CoursesStateType, WorkspaceCourseListType, WorkspaceCourseType} from '~/reducers/main-function/courses';
+import { loadMoreCoursesFromServer, LoadMoreCoursesFromServerTriggerType } from '~/actions/main-function/courses';
+
+
+interface WorkspacesProps {
+  i18n: i18nType,
+  coursepickerCoursesState: CoursesStateType,
+  coursepickerHasMore: boolean,
+  loadMoreCoursesFromServer: LoadMoreCoursesFromServerTriggerType,
+  coursepickerCoursesCourses: WorkspaceCourseListType
+}
+
+interface WorkspacesState {
+}
+
+class Workspaces extends React.Component<WorkspacesProps, WorkspacesState> {
+  render(){
+    if (this.props.coursepickerCoursesState === "LOADING"){
+      return null;
+    } else if (this.props.coursepickerCoursesState === "ERROR"){
+      //TODO: put a translation here please! this happens when messages fail to load, a notification shows with the error
+      //message but here we got to put something
+      return <div className="empty"><span>{"ERROR"}</span></div>
+    } else if (this.props.coursepickerCoursesCourses.length === 0){
+      return <div className="empty"><span>{this.props.i18n.text.get("plugin.coursepicker.searchResult.empty")}</span></div>
+    }
+    return (
+        <ApplicationList>
+        {this.props.coursepickerCoursesCourses.map((workspace: WorkspaceCourseType)=>{
+          return <Workspace key={workspace.id} course={workspace}/>
+        })}
+        {this.props.coursepickerCoursesState === "LOADING_MORE" ? <ApplicationListItem className="loader-empty"/> : null}
+        </ApplicationList>
+    );
+  }
+}
+
+function mapStateToProps(state: StateType){
+  return {
+    i18n: state.i18n,
+    coursepickerCoursesState: state.courses.state,
```

Comment:
We might not want to have states named coursepicker if they are not in coursepicker

Reply:
Did we ever decide to separate the standard courses vs the organization courses?

Reply:
> Did we ever decide to separate the standard courses vs the organization courses?

We did, it's happening in the other branch